src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/internal/utils/MergeUtils.kt

```diff
@@ -0,0 +1,105 @@
+package it.krzeminski.snakeyaml.engine.kmp.internal.utils
+
+import it.krzeminski.snakeyaml.engine.kmp.nodes.MappingNode
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Node
+import it.krzeminski.snakeyaml.engine.kmp.nodes.NodeTuple
+import it.krzeminski.snakeyaml.engine.kmp.nodes.ScalarNode
+import it.krzeminski.snakeyaml.engine.kmp.nodes.SequenceNode
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Tag.Companion.MERGE
+
+internal abstract class MergeUtils {
```

Comment:
I would make it a final class and pass the `asMappingNode` as a lambda via the constructor. It would make it a bit cleaner, I think

Reply:
Good idea, let me go with just this suggestion and skip the others so that, as you wrote, the code doesn't diverge too much from upstream for easier porting.